lib/core/jdl_application_configuration.js

```diff
@@ -81,12 +83,19 @@ function getFormattedConfigOptionsString(options, indent) {
     .join('\n');
 }
 
-function filterOutUnwantedOptions(options) {
-  return filterOutOptionsThatShouldNotBeExported(filterOutOptionsWithoutValues(options));
+function filterOutUnwantedOptions(options, exportOptions = {}) {
+  const { exportNotRegisteredConfigs } = exportOptions;
+  options = { ...options };
+  options = filterOutOptionsWithoutValues(options);
+  options = filterOutOptionsThatShouldNotBeExported(options);
+  if (!exportNotRegisteredConfigs) {
+    options = filterOutGenericOptions(options, exportOptions);
```

Comment:
The function takes one argument instead of two.

Reply:
```suggestion
    options = filterOutGenericOptions(options);
```